src/olympia/files/tests/test_utils_.py

```diff
@@ -1189,10 +1189,49 @@ def test_id_to_path(value, expected):
 
 
 class TestSafeZip(TestCase):
+    def test_raises_error_for_invalid_webextension_xpi(self):
+        with pytest.raises(forms.ValidationError):
+            utils.SafeZip(get_addon_file('invalid_webextension.xpi'))
 
     def test_raises_validation_error_when_uncompressed_size_is_too_large(self):
         with override_settings(MAX_ZIP_UNCOMPRESSED_SIZE=1000):
             with pytest.raises(forms.ValidationError):
                 # total uncompressed size of this xpi is: 2269 bytes
                 utils.SafeZip(get_addon_file(
                     'valid_firefox_and_thunderbird_addon.xpi'))
+
+
+class TestArchiveMemberValidator(TestCase):
+    # We cannot easily test `archive_member_validator` so let's test
+    # `_validate_archive_member_name_and_size` instead.
+
+    def test_raises_when_filename_is_none(self):
+        with pytest.raises(forms.ValidationError):
+            utils._validate_archive_member_name_and_size(None, 123)
+
+    def test_raises_when_filesize_is_none(self):
+        with pytest.raises(forms.ValidationError):
+            utils._validate_archive_member_name_and_size('filename', None)
+
+    def test_raises_when_filename_is_dot_dot_slash(self):
+        with pytest.raises(forms.ValidationError):
+            utils._validate_archive_member_name_and_size('../', 123)
+
+    def test_raises_when_filename_starts_with_slash(self):
+        with pytest.raises(forms.ValidationError):
+            utils._validate_archive_member_name_and_size('/..', 123)
+
+    def test_raises_when_filename_is_dot_dot(self):
+        with pytest.raises(forms.ValidationError):
+            utils._validate_archive_member_name_and_size('..', 123)
+
+    def test_does_not_raise_when_filename_is_dot_dot_extension(self):
+        utils._validate_archive_member_name_and_size('foo..svg', 123)
+
+    @override_settings(FILE_UNZIP_SIZE_LIMIT=100)
+    def test_raises_when_filesize_is_above_limit(self):
+        with pytest.raises(forms.ValidationError):
+            utils._validate_archive_member_name_and_size(
+                'filename',
+                settings.FILE_UNZIP_SIZE_LIMIT + 100
+            )
```

Comment:
You could be using `@pytest.mark.parametrize` to simplify this test a bit and make it easier to see the actual scenarios and look through them imo (e.g https://github.com/mozilla/addons-server/blob/af81faedd5aab40c8674c76a7dc8e8670b69e5d5/src/olympia/lib/tests/test_git.py#L1061-L1109)

For using parametrize together with `pytest.raises` you could take a look at http://doc.pytest.org/en/latest/example/parametrize.html#parametrizing-conditional-raising